### PR TITLE
Remove deprecated functions and arguments for the 0.19 release

### PR DESCRIPTION
**Pull request by grlee77**
## Description

This PR implements the remaining deprecations listed under the 0.19 section of `TODO.txt`. These are broken up into separate small commits for easier reviewing.

## Checklist

- [Docstrings for all functions](https://github.com/numpy/numpy/blob/master/doc/example.py)
- Gallery example in `./doc/examples` (new features only)
- Benchmark in `./benchmarks`, if your changes aren't covered by an
  existing benchmark
- Unit tests
- Clean style in [the spirit of PEP8](https://www.python.org/dev/peps/pep-0008/)

## For reviewers

- Check that the PR title is short, concise, and will make sense 1 year
  later.
- Check that new functions are imported in corresponding `__init__.py`.
- Check that new features, API changes, and deprecations are mentioned in
      `doc/release/release_dev.rst`.

### CODE REVIEW

**Comment by grlee77** [comment]
Implemented a couple of remaining deprecations missing from the TODO as found by `grep -rI "0.19"` in the codebase.

Also fixed f2878e5 by adding the removal of `draw.circle` (forgot to save and commit that change previously!)

File: skimage/segmentation/slic_superpixels.py [hunk 1]
```diff
@@ -213,10 +213,9 @@ def slic(image, n_segments=100, compactness=10., max_num_iter=10, sigma=0,
       interpret them as 3D with the last dimension having length 3, use
       `channel_axis=-1`.
 
-    * `start_label` is introduced to handle the issue [4]_. The labels
-      indexing starting at 0 will be deprecated in future versions. If
-      `mask` is not `None` labels indexing starts at 1 and masked area
-      is set to 0.
+    * `start_label` is introduced to handle the issue [4]_. Labeling indexing
```
Comment (mkcor) [inline skimage/segmentation/slic_superpixels.py:216]:
```suggestion
    * `start_label` is introduced to handle the issue [4]_. Label indexing
```

File: skimage/segmentation/slic_superpixels.py [hunk 2]
```diff
@@ -213,10 +213,9 @@ def slic(image, n_segments=100, compactness=10., max_num_iter=10, sigma=0,
...
+      starts at 1 by default. The labels indexing starting at 0 will be
```
Comment (mkcor) [inline skimage/segmentation/slic_superpixels.py:217]:
```suggestion
      starts at 1 by default. The labels' index starting at 0 will be
```

File: TODO.txt [hunk 3]
```diff
@@ -8,44 +8,6 @@ Version 0.17
...
-
-Version 0.19
-------------
-
-* Set ``start_label`` to 1 in ``slic`` and remove the DeprecationWarning.
-* In ``skimage/draw/draw.py`` remove ``circle'' and related tests.
-* In ``skimage/segmentation/morphsnakes.py`` remove ``circle_level_set'' and related tests.
-  * Also make sure to look in the function ``_init_level_set``
-* In ``skimage/morphology/_flood_fill.py`` replace the deprecated parameter
-  in flood_fill() ``inplace`` with ``in_place`` and update the tests.
```
Comment (mkcor) [inline TODO.txt:23]: I didn't follow when or why this change was decided on... [pandas](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.drop.html) speaks 'inplace' so it seems more standard to me. Oh well.

**Review by mkcor — APPROVED** [review]

File: TODO.txt (hunk 3 above)
Comment (grlee77) [reply TODO.txt:23]: I don't remember, but I think it was for consistency with several other functions that use `in_place`. It looks like a number of functions under `future/graph` already have `in_place`, also `morphology.remove_small_holes` and `morphology.remove_small_objects.

Comment (grlee77) [reply TODO.txt:23]: That said, most of those are still in future, so changes are probably expected there. I wonder if there are other prominent examples of `inplace` aside from pandas? 

I don't think SciPy or NumPy tend to use either, usually requiring and `out` or `output` argument to be specified instead.

**Comment by grlee77** [comment]
Thanks for reviewing this fairly long PR @mkcor!

File: TODO.txt (hunk 3 above)
Comment (mkcor) [reply TODO.txt:23]: > I don't remember, but I think it was for consistency with several other functions that use `in_place`. It looks like a number of functions under `future/graph` already have `in_place`

Ok.

> also `morphology.remove_small_holes` and `morphology.remove_small_objects`

Right, but it looks like the argument is going away altogether for 1.0!
https://github.com/scikit-image/scikit-image/blob/7b0d37d4a3a3282a09b89175c0ee9197efb80b60/skimage/morphology/misc.py#L51
And going NumPy-style instead:
https://github.com/scikit-image/scikit-image/blob/7b0d37d4a3a3282a09b89175c0ee9197efb80b60/skimage/morphology/misc.py#L52

File: skimage/util/tests/test_arraypad.py [hunk 4]
```diff
@@ -2,19 +2,12 @@
 
 import numpy as np
```
Comment (rfezzani) [inline skimage/util/tests/test_arraypad.py:3]: Do we need to keep these tests for `np.pad`?

File: skimage/segmentation/slic_superpixels.py [hunk 5]
```diff
@@ -213,10 +213,9 @@ def slic(image, n_segments=100, compactness=10., max_num_iter=10, sigma=0,
       interpret them as 3D with the last dimension having length 3, use
       `channel_axis=-1`.
 
-    * `start_label` is introduced to handle the issue [4]_. The labels
-      indexing starting at 0 will be deprecated in future versions. If
-      `mask` is not `None` labels indexing starts at 1 and masked area
-      is set to 0.
+    * `start_label` is introduced to handle the issue [4]_. Label indexing
+      starts at 1 by default. The labels' index starting at 0 will be
+      deprecated in future versions.
```
Comment (rfezzani) [inline skimage/segmentation/slic_superpixels.py:218]: Isn't it already deprecated?
```suggestion
      starts at 1 by default.
```

**Review by rfezzani — COMMENTED** [review]
Thank you @grlee77, just left 2 minor suggestions.

File: skimage/util/tests/test_arraypad.py (hunk 4 above)
Comment (grlee77) [reply skimage/util/tests/test_arraypad.py:3]: Yeah, I don't see why we need to keep these. If there are deficiencies in NumPy's `pad` tests we should just improve them there.

Comment (grlee77) [reply skimage/util/tests/test_arraypad.py]: I have now removed this file.

**Review by rfezzani — APPROVED** [review]
Excellent, thank you @grlee77. just waiting for the green CI before merge :wink:

**Pull request merged**